eclipse-cla/src/main/java/org/eclipse/foundation/gerrit/validation/EclipseCommitValidationListener.java

```diff
@@ -149,104 +178,158 @@ public EclipseCommitValidationListener(PluginConfigFactory cfgFactory) {
+		 * Only committers can push on behalf of other users. Note that, I am asking if
+		 * the user (i.e. the person who is doing the actual push) is a committer.
 		 */
-		if (!author.isPresent() || !author.get().getAccount().equals(user.getAccount())) {
-			if (!isCommitter(user, project)) {
-				messages.add(new CommitValidationMessage("You are not a project committer.", true));
-				messages.add(new CommitValidationMessage("Only project committers can push on behalf of others.", true));
-				addDocumentationPointerMessage(messages);
-				addEmptyLine(messages);
-				throw new CommitValidationException("You must be a committer to push on behalf of others.", messages);	
-			}
-		} 
+		if ((!author.isPresent() || !author.get().getAccount().equals(user.getAccount()))
+				&& (!isCommitter(user, project) && (!author.isPresent()
+						|| !getPreviousPatchsetAuthors(project, commit).contains(author.get())))) {
+			messages.add(new CommitValidationMessage("You are not a project committer.", true));
+			messages.add(new CommitValidationMessage("Only project committers can push on behalf of others.", true));
+			addDocumentationPointerMessage(messages);
+			addEmptyLine(messages);
+			throw new CommitValidationException("You must be a committer to push on behalf of others.", messages);
+		}
 
 		messages.add(new CommitValidationMessage("This commit passes Eclipse validation.", false));
-		
+
 		return messages;
 	}
 
+	private Set<IdentifiedUser> getPreviousPatchsetAuthors(Project project, RevCommit commit) {
+		Set<IdentifiedUser> authors = new HashSet<>();
+		try {
+			
+			ChangeInfo info = null;
+			// for each available Change ID (there should be one) attempt to get change info 
+			List<String> changeIds = commit.getFooterLines(FooterConstants.CHANGE_ID);
+			for (String changeId :changeIds) {
+				List<ChangeInfo> cis = this.gapi.changes().query(changeId).get();
+				// if we have results, set back and break out
+				if (!cis.isEmpty()) {
+					info = cis.get(0);
+					break;
+				}
+			}
+			
+			// if we found a valid change info using the footer lines, continue processing
+			if (info != null) {
+				// get the control object for the current change set
+				ChangeData cd = this.cdf.create(this.db.get(), project.getNameKey(), new Change.Id(info._number));
+				ChangeControl changeControl = cd.changeControl();
+				try (Repository repo = this.repoManager.openRepository(project.getNameKey());
+						RevWalk rw = new RevWalk(repo)) {
+					ImmutableCollection<PatchSet> patchSets = this.patchsetUtil.byChange(this.db.get(), changeControl.getNotes());
+					for (PatchSet p : patchSets) {
+						AnyObjectId commitId = ObjectId.fromString(p.getRevision().get());
+						RevCommit c = rw.parseCommit(commitId);
+						if (c.equals(commit)) {
+							continue;
+						}
+						Optional<IdentifiedUser> u = identifyUser(c.getAuthorIdent());
+						if (u.isPresent()) {
+							authors.add(u.get());
+						}
+					}
+				}
+			}
```

Comment:
Please add some DEBUG logging on the `else` case to make it easier to diagnose / test